### DISCUSSION TITLE
Product image upload missing file extension in customization fields

### DISCUSSION THREAD

**Discussion by hamdiayariprestashop** [PrestaShop/PrestaShop#36973] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In **ProductController.php**, the **pictureUpload()** function is responsible for handling image uploads when a customer customizes a product. The issue occurs because this function generates a random file name using md5(uniqid()) but does not append the original file extension (e.g., .jpg, .png). As a result, the image file is saved without an extension in the upload folder. This leads to problems when attempting to display or retrieve the file later, as the missing extension prevents browsers from correctly interpreting the file type. Here is a link from the PrestaShop forums, where some people have reported a similar issue since 2015: [Prestashop Forums](https://www.prestashop.com/forums/topic/393872-product-customization-field-image-upload-doesnt-work/)
Attached below is a capture of the infected code and the FTP upload results : 

![Capture](https://github.com/user-attachments/assets/8da06e35-fe9d-4ac0-b49b-d0c820bbf2d6)

![Capture2](https://github.com/user-attachments/assets/e28bf983-1e03-4a12-9899-f3febde64bb7)

### Expected behavior

When an image is uploaded for product customization, the uploaded file should retain its original file extension (e.g., .jpg, .png, .gif). The file should be saved in the **upload** directory with the correct extension so it can be correctly interpreted by browsers and other services when retrieved later.

### Steps to reproduce

**Go to the product page:**

1. Navigate to a product that allows customization (i.e., product with file upload option).
2. Upload a file: Use the file upload option to upload an image (e.g., .jpg, .png, or .gif).
3. Complete the checkout process: Add the customized product to your cart, and complete the order.
4. Check the server directory: Access the /upload folder via FTP or the server file manager.
5. Observe the issue: Notice that the uploaded file is saved without an extension, making it difficult to retrieve or display.

### PrestaShop version(s) where the bug happened

8.1.7

### PHP version(s) where the bug happened

8.1.25

**Comment by florine2623** [comment]
Hello @hamdiayariprestashop ,

I transfered this to Discussions, as it is not an issue :)

Thanks

**Comment by melikslab** [comment]
Hi there!
I encountered a similar issue with file access in the upload directory, and the solution is straightforward. Modify the .htaccess file in your uploads folder with the following configuration:
```
apache# Allow access to files
<IfModule mod_authz_core.c>
    Require all granted
</IfModule>
# Prevent directory listing
Options -Indexes


```
Hope this helps!

**Comment by rpereira3** [comment]
Hi there :)

I’m working on PrestaShop 8.2.2, and I’m also experiencing this issue.

@florine2623, the fact that the file doesn’t have an extension is not an issue. However, I think that the image being visualy broken (not accessible) by default is one.

Is @melikslab's fix the right solution? If so, would it be possible to fix it directly into PrestaShop core ?

**Reply by kpodemski** [reply]
Hello @rpereira3 

You can find the fix here: https://github.com/PrestaShop/PrestaShop/pull/40262

**Comment by kpodemski** [comment]
It is done intentionally to ensure that the MIME type is checked in real-time for security purposes. If anyone is having trouble displaying customization images on PrestaShop version 8.1.6 or later, the fix can be found here: https://github.com/PrestaShop/PrestaShop/pull/40262
